pennylane/devices/null_qubit.py

```diff
@@ -129,6 +132,48 @@ def _interface(config: ExecutionConfig):
     return config.interface.get_like() if config.gradient_method == "backprop" else "numpy"
 
 
+def _simulate_resource_use(circuit):
+    RESOURCES_FNAME = "__pennylane_resources_data.json"
+
+    num_wires = len(circuit.wires)
+    gate_types = defaultdict(int)
+
+    for op in circuit.operations:
+        name = ""
+        while hasattr(op, "base"):
+            if type(op) in (Controlled, ControlledOp):
+                # Don't check this with `isinstance` to avoid unrolling ops like CNOT
+                name += "C_"
+            elif isinstance(op, Adjoint):
+                name += "adj_"
+            else:
+                break  # Certain gates have "base" but shouldn't be broken down (like CNOT)
+            # TODO: How should these be handled?
+            # if isinstance(op, (Pow, Exp)):
+            #     op = op.base
```

Comment:
There are no ways to handle these ops in Catalyst Runtime :thinking:

Reply:
Happy with not handling them for now! Pending @AntonNI8's approval

Reply:
Left a comment that notes this in [f09cf93](https://github.com/PennyLaneAI/pennylane/pull/7226/commits/f09cf93955d90461d07e37bfb73c812743c3c40c)